From 9e939c93dbd88efb3e7e221bcf7cc0fb934ca465 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:46:14 -0400
Subject: [pci] Clarify the "scan anyway" comment in pci_scan_bridge()

Message-id: <20140806034614.1438.76949.stgit@gir.stowe>
Patchwork-id: 87429
O-Subject: [RHEL7.1 PATCH 20/69] PCI: Clarify the "scan anyway" comment in pci_scan_bridge()
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 2ed8582341f651ca14d00ab0ada4b46f493e1fcb

commit 2ed8582341f651ca14d00ab0ada4b46f493e1fcb
Author: Andreas Noever <andreas.noever@gmail.com>
Date:   Thu Jan 23 21:59:22 2014 +0100

    PCI: Clarify the "scan anyway" comment in pci_scan_bridge()

    Initially when we encountered a bus that was already present we skipped
    it. Since 74710ded8e16 'PCI: always scan child buses' we continue
    scanning in order to allow user triggered rescans of already existing
    busses.

    The old comment suggested that the reason for continuing the scan is a
    bug in the i450NX chipset. This is not the case.

    Signed-off-by: Andreas Noever <andreas.noever@gmail.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index ddd7096..9324a44 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -848,11 +848,10 @@ int pci_scan_bridge(struct pci_bus *bus, struct pci_dev *dev, int max, int pass)
    goto out;
 
   /*
-   * If we already got to this bus through a different bridge,
-   * don't re-add it. This can happen with the i450NX chipset.
-   *
-   * However, we continue to descend down the hierarchy and
-   * scan remaining child buses.
+   * The bus might already exist for two reasons: Either we are
+   * rescanning the bus or the bus is reachable through more than
+   * one bridge. The second case can happen with the i450NX
+   * chipset.
    */
   child = pci_find_bus(pci_domain_nr(bus), secondary);
   if (!child) {
-- 
1.7.1